flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPool.java

```diff
@@ -190,6 +205,12 @@
 			subpartitionBufferRecyclers[i] = new SubpartitionBufferRecycler(i, this);
 		}
 		this.maxBuffersPerChannel = maxBuffersPerChannel;
+
+		if (checkAvailability()) {
+			availabilityHelper.resetAvailable();
+		}
+
+		assert hasConsistentAvailability();
```

Comment:
Similar to `NetworkBufferPool#redistributeBuffers` these assertions are just to ease debugging during test execution. The few cycles needed to conduct the test are skipped on production. If you rather like to see them being executed for every run, we probably need to align it also in `NetworkBufferPool` to keep it consistent.

Reply:
I mean I do not see a performance benefit of avoiding a couple of `checkState`s on per buffer path, while it could provide some help, if bug will occur on some production setup only.

In per record paths, I guess I would agree more with `assert`s